types/views/views.go

```diff
@@ -44,83 +44,77 @@ func (v *StringSlice) UnmarshalJSON(b []byte) error {
 }
 
 // IsNil reports whether the underlying slice is nil.
-func (v StringSlice) IsNil() bool { return v.ж == nil }
+func (v Slice[T]) IsNil() bool { return v.ж == nil }
 
 // Len returns the length of the slice.
-func (v StringSlice) Len() int { return len(v.ж) }
+func (v Slice[T]) Len() int { return len(v.ж) }
 
 // At returns the string at index `i` of the slice.
-func (v StringSlice) At(i int) string { return v.ж[i] }
+func (v Slice[T]) At(i int) T { return v.ж[i] }
 
 // AppendTo appends the underlying slice values to dst.
-func (v StringSlice) AppendTo(dst []string) []string {
+func (v Slice[T]) AppendTo(dst []T) []T {
 	return append(dst, v.ж...)
 }
 
 // AsSlice returns a copy of underlying slice.
-func (v StringSlice) AsSlice() []string {
+func (v Slice[T]) AsSlice() []T {
 	return v.AppendTo(v.ж[:0:0])
 }
 
 // IPPrefixSlice is a read-only accessor for a slice of netaddr.IPPrefix.
 type IPPrefixSlice struct {
-	// It is named distinctively to make you think of how dangerous it is to escape
-	// to callers. You must not let callers be able to mutate it.jd
-	ж []netaddr.IPPrefix
+	ж Slice[netaddr.IPPrefix]
```

Comment:
Did you try embedding this so you wouldn't need all the forwarding accessors like IsNil, Len, At, etc?

Reply:
I did, didn't work.

Reply:
Sad. Oh well.